### PR TITLE
Removing `scheduler_factory`

### PR DESCRIPTION
**Pull request by lookfwd**
Replacing `scheduler_factory` with a module discovery mechanism. Still not crazy modular but way more extensible.

### CODE REVIEW

File: qtopconf.yaml [hunk 1]
```diff
@@ -64,6 +66,18 @@ state_abbreviations:
     s: suspended_of_user
     S: suspended_by_the_queue
     shite: testing_of_user
+  pbs:
```
Comment (lookfwd) [inline qtopconf.yaml]: Would be nice if all those ware going back to individual sources with a mechanism similar to `get_mnemonic():`.

File: qtopconf.yaml [hunk 2]
```diff
@@ -22,6 +22,8 @@ schedulers:
     oarstat_file: %(savepath)s/oarstat%(pid)s.txt, oarstat
   sge:
     sge_file_stat: %(savepath)s/qstat%(pid)s.F.xml.stdout, qstat -F -xml -u '*'
+  demo:
```
Comment (lookfwd) [inline qtopconf.yaml]: Would be nice if all those ware going back to individual sources with a mechanism similar to `get_mnemonic():`.

File: qtopconf.yaml (hunk 1 above)
Comment (sfranky) [inline qtopconf.yaml]: Not 100% sure, but my impression is @fgeorgatos wanted them to be customiseable by the user

File: qtopconf.yaml (hunk 2 above)
Comment (sfranky) [inline qtopconf.yaml]: These *need* to be user customiseable!

**Comment by lookfwd** [comment]
Will redo on dev branch

**Comment by lookfwd** [comment]
See #141

**Pull request closed without merging**